File: go/gin/internal/modules/auth/controller.go
package auth

import (
	"net/http"
	"tcc/internal/common/errors"
	"tcc/internal/common/helper"
	"tcc/internal/common/messages"
	"tcc/internal/modules/auth/dto"

	"github.com/gin-gonic/gin"
)

type AuthController struct{
	service AuthService
}


func (c *AuthController) LoginClient(ctx *gin.Context){
	var dto dto.LoginAuthDto
	if err := ctx.ShouldBindJSON(&dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err.Error())
		ctx.JSON(badRequest.StatusCode, badRequest)
		return
	}
	
	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	response, err := c.service.LoginClient(dto)

	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.JSON(http.StatusOK, response)
}

func (c *AuthController) LoginManager(ctx *gin.Context){
	var dto dto.LoginAuthDto
	if err := ctx.ShouldBindJSON(&dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err.Error())
		ctx.JSON(badRequest.StatusCode, badRequest)
		return
	}
	
	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	response, err := c.service.LoginManager(dto)

	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.JSON(http.StatusOK, response)
}

File: rust/actix/src/modules/auth/controller.rs
use actix_web::{web, HttpResponse, Responder};
use crate::common::errors::base::BaseError;
use crate::common::errors::bad_request::bad_request_error;
use crate::common::messages::error::ErrorMessages;
use validator::Validate;

use super::service::AuthService;
use super::dto::LoginAuthDto;

#[derive(Debug)]
pub struct AuthController;

impl AuthController {
    pub async fn login_client(
        dto: web::Json<LoginAuthDto>,
    ) -> Result<impl Responder, BaseError> {
        if let Err(errors) = dto.validate() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(errors)));
        }

        match AuthService::login_client(dto.into_inner()).await {
            Ok(result) => Ok(HttpResponse::Ok().json(result)),
            Err(err) => Err(err),
        }
    }

    pub async fn login_manager(
        dto: web::Json<LoginAuthDto>,
    ) -> Result<impl Responder, BaseError> {
        if let Err(errors) = dto.validate() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(errors)));
        }

        match AuthService::login_manager(dto.into_inner()).await {
            Ok(result) => Ok(HttpResponse::Ok().json(result)),
            Err(err) => Err(err),
        }
    }
}